org.eclipse.scout.rt.api.data/src/main/java/org/eclipse/scout/rt/api/data/security/PermissionDo.java

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright (c) 2010, 2023 BSI Business Systems Integration AG
+ *
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.eclipse.scout.rt.api.data.security;
+
+import javax.annotation.Generated;
+
+import org.eclipse.scout.rt.dataobject.DoEntity;
+import org.eclipse.scout.rt.dataobject.DoValue;
+import org.eclipse.scout.rt.dataobject.TypeName;
+import org.eclipse.scout.rt.dataobject.TypeVersion;
+import org.eclipse.scout.rt.security.IPermission;
+
+/**
+ * <b>IMPORTANT</b>: Never persist this data object or add a {@link TypeVersion}.
+ */
+@TypeName("scout.Permission")
+public class PermissionDo extends DoEntity {
+
+  public PermissionDo() {
+    withObjectType("Permission");
+  }
+
+  /**
+   * String used to identify the Scout JS object created by this object model.
+   */
+  public DoValue<String> objectType() {
```

Comment:
ScoutJS-specific type as part of API? Should the ScoutJS UI not be able to determine the correct JS class based on typename of DO? Actually DO subclass implementations seems to match 1:1 to a ScoutJS class